### ISSUE TITLE
Add ability to post message to Webhook URL's

### ISSUE THREAD

**Issue by gep13** [uwp-squad/gitter-api-pcl#4]
In order to see messages appear in the Activity Feed of a Gitter Room, after creating a Custom Webhook.

**Comment by Odonno** [comment]
Based on HTTP REST API, could you describe precisely how to create webhook methods ?

I suppose it is coming from https://developer.gitter.im/docs/services, but I have no experience on it.

**Comment by gep13** [comment]
That is correct, there is no API method to post to a Custom Webhook, as it doesn't require authentication. you can see the code that I am using here:

https://github.com/gep13/Cake.Gitter/blob/develop/Source/Cake.Gitter/Chat/GitterChatApi.cs#L82

The information on how to get the Webhook URL can be found here:

https://github.com/gep13/Cake.Gitter#web-hook-url

**Comment by Odonno** [comment]
It is not completely clear in my mind at this moment. When you get time, could you try something to clear this up ? i think there is some lacks of information : 
1. How to retrieve webhook url ? I suppose it's not based on room id, but another random id ?
2. How do you use this webhook ? What are the directions ? (from Gitter to .., from .. to Gitter, both ways ?)
3. What is the data you are sending ?

**Comment by gep13** [comment]
1. Yes, this is random.  The instructions on the second link that I posted above show how you can retrieve this.  You get it once you set up a Custom Webhook Integration within the Gitter Settings for your Room
2. It is one way.  You simply POST an Http Payload that contains a "message", as shown in the first link above
3. See answer to 2.

**Comment by Odonno** [comment]
@gep13 So, if I understand it well, it is simply a POST method to Gitter API (using `webhook url` and `message` payload, see that `level` is optional).

Can you confirm that it looks like that in code : 

```
public Task PostWebhook(string url, string content, MessageLevel? level = null)
{
    // Create an HttpClient and send content payload
    string payload = $"message={content}";
}
```

Or do you receive a message from this POST method ?

**Comment by gep13** [comment]
Exactly that, here is the code that I currently use:

```
            using (var client = new WebClient())
            {
                client.Headers[HttpRequestHeader.ContentType] = "application/x-www-form-urlencoded";

                var postBytes = Encoding.UTF8.GetBytes(string.Format("message={0}", message));

                var resultBytes = client.UploadData(
                        messageSettings.IncomingWebHookUrl,
                        "POST",
                        postBytes);

                var result = Encoding.UTF8.GetString(resultBytes);

                var parsedResult = new GitterChatMessageResult(
                    StringComparer.OrdinalIgnoreCase.Equals(result, "ok"),
                    DateTime.UtcNow.ToString("u"),
                    StringComparer.OrdinalIgnoreCase.Equals(result, "ok") ? string.Empty : result);

                context.Debug("Result parsed: {0}", parsedResult);

                return parsedResult;
            }
```

**NOTE:** Taken from the link above

I am using WebClient, rather than HttpClient, but there should be nothing to stop you using that.  There is a very basic response returned, which is "ok" if the post happened, and there will be an error if it failed.

**Comment by Odonno** [comment]
Ok, so, we could go for a `bool` returned value ?

```
public Task<bool> PostWebhook(string url, string content, MessageLevel? level = null)
{
    string payload = $"message={content}";
    // TODO : Create an HttpClient and send content payload    

    return result == "ok";
}
```

**Comment by gep13** [comment]
yip, that would seem appropriate to me :+1:

**Comment by Odonno** [comment]
@gep13 It's now available in the version 0.4.0. Really thanks for your help, I could not do it without you !

**Comment by gep13** [comment]
Nice one, I will take a look, thanks! :+1:
